8.cytopick_analysis/scripts/prediction_images_find.py

```diff
@@ -0,0 +1,532 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# This notebook finds random cells from each prediction category and displays them. The purpose is to get representative images examples of each category.
+
+# In[1]:
+
+
+import pathlib
+
+import cv2
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+
+# import pillow and open cv
+import PIL
+import seaborn as sns
+import tifffile as tf
+from cytocherrypick import cherrypick
+from PIL import Image, ImageEnhance
+from tqdm import tqdm
+
+# In[2]:
+
+
+# function that selects a random image from the dataframe
+def random_cell_select(
+    df: pd.DataFrame,
+    n: int = 1,
+) -> pd.DataFrame:
+    """
+    Selects a random cell from the dataframe
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+        Dataframe containing the cell features
+    n : int, optional
+        Number of random cells to select, by default 1
+
+    Returns
+    -------
+    pd.DataFrame
+        The return dataframe with the random cell selected
+    """
+
+    # select a random cell
+    random_cell = df.sample(n=n, random_state=0)
+    return random_cell
+
+
+# In[3]:
+
+
+# parameters
+CELL_TYPE = "PBMC"
+feature = "Nuclei_Texture_SumVariance_CorrGasdermin_3_01_256"
+
+
+# In[4]:
+
+
+# define directories
+# where the images are
+image_dir_path = pathlib.Path(
+    "/media/lippincm/18T/interstellar_data/70117_20230210MM1_Gasdermin514_CP_BC430856__2023-03-22T15_42_38-Measurement1/2.IC/"
+)
+# if path does not exist, create it
```

Comment:
Same comment as other file regarding my confusion of making an empty dir for images.

Reply:
Switched out for a `.absolute(resolve=True)` statement instead